capstone-frontend/src/components/ClubGridItem.jsx

```diff
@@ -32,12 +33,38 @@ class ClubGridItem extends Component {
         .catch(e => console.log(e));
   }
 
+  fetchBook = () => {
+    if (this.props.club.gbookID.length === 0) {
+      this.setState({ title: 'Nothing yet!' });
+    } else {
+      fetch(`/api/search?gbookId=${this.props.club.gbookID}`)
+        .then(response => response.json())
+        .then(books => books[0])
+        .then(book => book.authors && book.authors.length > 0 
+          ? book.title
+          : 'Nothing yet!'
+        )
+        .then(title => this.setState({ title }))
+        .catch(e => {
+          console.error(e);
+          return 'Nothing yet!';
+        });
+    }
+  }
+
+  componentDidMount() {
+    this.fetchBook();
+  }
+
   render() {
+    console.log(this.props.club);
     const userID = window.localStorage.getItem("userID");
     const isOwner = this.props.club.ownerID === userID
     const isMember = this.props.club.memberIDs.includes(userID);
     const isViewer = !isOwner && !isMember;
     const isRequester = this.props.club.requestIDs.includes(userID);
+    const title = this.state.title && this.state.title.length > 0 ?
+      this.state.title : 'Nothing yet!';
```

Comment:
nit: Since this string literal is used several times throughout this file it might be good to store it in a DEFAULT_TITLE constant

Reply:
Good idea, I'll change this.